### PR TITLE
ssc mappings

### PR DESCRIPTION
**Pull request by FrankMittelbach**
# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [n/a] Test file(s) added    --- (not testable with the fonts we have in the test suite)
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [x] Rollback provided (if necessary)?
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: base/ltfssaxes.dtx [hunk 1]
```diff
@@ -1175,14 +1177,25 @@
...
 %
-%    So far there aren't any rules for it---probably there should be
-%    some putting it in the same category as \texttt{sc}.
+%    Michael Ummels kindly prepared the necessary rules for
+%    \texttt{ssc} ages ago, but until recently they managed to hide
+%    deep down in my inbox. I
+%    finally got around integrating them (with a few changes) and also
+%    took the opportunity to rationalize the other uncommon
+%    \texttt{sw} shape rules a bit.
 %    \begin{macrocode}
```
Comment (car222222) [inline base/ltfssaxes.dtx:1199]: Rearrangement:
. . . to rationalize (a bit) the rules for the only other uncommon shape, \texttt{sw}.

File: base/ltfssaxes.dtx [hunk 2]
```diff
@@ -1195,59 +1208,110 @@
...
+%    assumptions: if \texttt{ssc}\meta{X} exists then
+%    \texttt{sc}\meta{X} and \meta{X} also exist in the font.  If the
+%    \texttt{ssc}\meta{X} shape doesn't exist but the user had asked
+%    for \texttt{ssc} we try to replace it by \texttt{sc} because we
+%    then assume that the current font family simply doesn't have any
+%    \texttt{ssc} shapes.  However, if we are already in some
+%    \texttt{ssc} shape and a change is requested we know that at
+%    least some \texttt{ssc} shapes exist for the current font family,
+%    so rather than falling back to some \texttt{sc} we try to stay
+%    within \texttt{ssc} changes in a fallback situation.
```
Comment (car222222) [inline base/ltfssaxes.dtx:1268]: ?? delete: changes ?

Not sure what is intended here??

File: base/ltfssaxes.dtx [hunk 3]
```diff
@@ -1195,59 +1208,110 @@
...
 %    \begin{macrocode}
 %\DeclareFontShapeChangeRule {it}{n}   {n}     {}
 %\DeclareFontShapeChangeRule {it}{it}  {it}    {}
 \DeclareFontShapeChangeRule {it}{sl}  {sl}    {it}
-%\DeclareFontShapeChangeRule {it}{sw}  {sw}    {}
+\DeclareFontShapeChangeRule {it}{sw}  {sw}    {it}
 %    \end{macrocode}
 %    If neither \texttt{scit} nor \texttt{scsl} exist then \texttt{sc}
 %    will be used as a fallback albeit with a log entry, so except for
 %    the latter there will be no change for CM or Latin Modern fonts.
```
Comment (car222222) [inline base/ltfssaxes.dtx:1283]: I have not checked carefully all the details in the code below.

Can do this later if required! 

One observation now:
Should the alignments of the arguments be tidied up?

File: base/ltfssaxes.dtx [hunk 4]
```diff
@@ -1195,59 +1208,110 @@
...
 \DeclareFontShapeChangeRule {sc}{it}  {scit}  {scsl}
 \DeclareFontShapeChangeRule {sc}{sl}  {scsl}  {scit}
-\DeclareFontShapeChangeRule {sc}{sw}  {scsw}  {sw}
+\DeclareFontShapeChangeRule {sc}{sw}  {scsw}  {scit}
 %\DeclareFontShapeChangeRule {sc}{sc}  {sc}    {}
-\DeclareFontShapeChangeRule {sc}{ulc} {n}     {}
+\DeclareFontShapeChangeRule {sc}{ulc}  {n}    {}
 %    \end{macrocode}
 %
 %    The next rule might be a bit surprising and rightly so. Correct
```
Comment (car222222) [inline base/ltfssaxes.dtx:1311]: Add comma:
. . . surprising, and rightly . . .

File: base/ltfssaxes.dtx [hunk 5]
```diff
@@ -1195,59 +1208,110 @@
...
 %    would be that \texttt{sc} is not affected by \texttt{up}, i.e.,
```
Comment (car222222) [inline base/ltfssaxes.dtx:1312]: Maybe:
It would be more correct if \text{sc}
were not . . . 

(or "was" if you don't like the subjunctive:-)

Comment (car222222) [inline base/ltfssaxes.dtx:1312]: And replace "i.e.," with "so that it"

File: base/ltfssaxes.dtx [hunk 6]
```diff
@@ -1262,34 +1326,39 @@
...
-\DeclareFontShapeChangeRule {scit}{sw}  {scsw}  {sc}    % or scit?
+\DeclareFontShapeChangeRule {scit}{sw}  {scsw}  {scit}
 \DeclareFontShapeChangeRule {scit}{sc}  {scit}  {}
 \DeclareFontShapeChangeRule {scit}{ulc} {it}    {}
-\DeclareFontShapeChangeRule {scit}{up} {sc}     {}
 %    \end{macrocode}
-%    The previous rule assumes that if \texttt{scit} exists then
+%    The next rule assumes that if \texttt{scit} exists then
 %    \texttt{it} exists as well. If not, the mechanism will save
 %    \texttt{ulc} in \cs{f@series} which most certainly doesn't
```
Comment (car222222) [inline base/ltfssaxes.dtx:1342]: Better: add comma before "which"

File: base/ltfssaxes.dtx [hunk 7]
```diff
@@ -1299,38 +1368,176 @@
...
+%<latexrelease>\DeclareFontShapeChangeRule {scsw}{sw}  {scsw}  {}
+%<latexrelease>\DeclareFontShapeChangeRule {scsw}{sc}  {scsw}  {}
+%<latexrelease>\DeclareFontShapeChangeRule {scsw}{ulc} {sw}    {}
+%<latexrelease>\DeclareFontShapeChangeRule {scsw}{up} {sc}     {}
+%<latexrelease>\DeclareFontShapeChangeRule {sw}{sc}  {scsw}  {}
+%<latexrelease>\DeclareFontShapeChangeRule {sw}{ulc} {sw}    {}
+%<latexrelease>\DeclareFontShapeChangeRule {sw}{up} {n}      {}
+%<latexrelease>
+%    \end{macrocode}
+%    In 2022-02-02 the \texttt{ssc} shape had no rules, so
```
Comment (car222222) [inline base/ltfssaxes.dtx:1482]: Better?:
. . . so that whenever . . .

File: base/ltfssaxes.dtx [hunk 8]
```diff
@@ -1299,38 +1368,176 @@
...
+%    whenever it was used it would have resulted in just switching to
```
Comment (car222222) [inline base/ltfssaxes.dtx:1483]: Even better! :
so that any use of it would . . .

File: base/ltfssaxes.dtx [hunk 9]
```diff
@@ -1299,38 +1368,176 @@
...
+%    that shape or switching away from it, so we mimic that by
```
Comment (car222222) [inline base/ltfssaxes.dtx:1484]: Either change the comma to ";" or start a new sentence here.

File: base/ltfssaxes.dtx [hunk 10]
```diff
@@ -1299,38 +1368,176 @@
...
+%    providing the corresponding trivial rules to overrule the ones
```
Comment (car222222) [inline base/ltfssaxes.dtx:1485]: the ones ==> those

**Review by car222222 — CHANGES_REQUESTED** [review]
Not sure whether I found everything here -- Strange stuff!

File: base/ltfssaxes.dtx (hunk 3 above)
Comment (FrankMittelbach) [reply base/ltfssaxes.dtx:1283]: they are sort of (if you remove the % on the lines that have one)

**Comment by FrankMittelbach** [comment]
> Not sure whether I found everything here -- Strange stuff!

well, yes. Not something many people will every dive into but something people just expect to work (which it does)

File: base/ltfssaxes.dtx (hunk 3 above)
Comment (FrankMittelbach) [reply base/ltfssaxes.dtx:1282]: I now went over all rules and tried to normalize/align their arguments to make them easier to compare.

**Review by josephwright — APPROVED** [review]

**Comment by FrankMittelbach** [comment]
Further alignment of arguments will now be handled in a different branch, which is why I backed theout here.

@car222222 I don't really expect you to re-read all the changes (done all of the ones you requested I think) but if you "request changes" it looks cleaner if you change this to approve eventually before I merge.

**Comment by car222222** [comment]
@FrankMittelbach OK, understood.

I shall finish off this one and review the new one, #1586, RSN (= over the weekend).

File: base/ltfssaxes.dtx [hunk 11]
```diff
@@ -1299,38 +1368,176 @@
...
+%\DeclareFontShapeChangeRule {ssc}{n}   {n}     {}
+\DeclareFontShapeChangeRule {ssc}{it}  {sscit}  {sscsl}
+\DeclareFontShapeChangeRule {ssc}{sl}  {sscsl}  {sscit}
+\DeclareFontShapeChangeRule {ssc}{sw}  {sscsw}  {sscit}
+\DeclareFontShapeChangeRule {ssc}{sc}  {sc}     {}
+%\DeclareFontShapeChangeRule {ssc}{ssc}  {ssc}    {}
+\DeclareFontShapeChangeRule {ssc}{ulc} {n}      {}
+%    \end{macrocode}
+%    We implement the same logic as for \texttt{sc}, see above. The
+%    \texttt{ssc} shape doesn 't to care for 30 years of history, but
```
Comment (car222222) [inline base/ltfssaxes.dtx:1396]: Something wrong here! Maybe it should be something like this:

. . . shape doesn't have/need to care about 30 years . . .

File: base/ltfssaxes.dtx [hunk 12]
```diff
@@ -1299,38 +1368,176 @@
...
+%    it would be surprising is \cs{sscshape}\cs{upshape} would work
```
Comment (car222222) [inline base/ltfssaxes.dtx:1397]: is ==> if

would work differently to ==>  did not work like

File: base/ltfssaxes.dtx (hunk 7 above)
Comment (car222222) [reply base/ltfssaxes.dtx:1482]: Not changed?

**Review by car222222 — APPROVED** [review]
Two more places that need attention, plus one question.

**Comment by car222222** [comment]
I shall wait on reviewing #1586 until the current round of changes there seems to have been completed.

**Comment by FrankMittelbach** [comment]
> I shall wait on reviewing #1586 until the current round of changes there seems to have been completed.

I'll fix the above in #1586 so that I don't get unnecessary merge conflicts there

**Pull request merged**